### PR TITLE
polkit: 0.119 → 0.120

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
https://lists.freedesktop.org/archives/polkit-devel/2021-October/000608.html

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @Binary-Eater

### CODE REVIEW

File: pkgs/development/libraries/polkit/default.nix [hunk 1]
```diff
@@ -12,19 +33,23 @@
 }:
 
 let
-
+  boolToString = b: if b then "true" else "false";
```
Comment (Binary-Eater) [inline pkgs/development/libraries/polkit/default.nix:36]: Is it possible to use [`lib.boolToString`](https://github.com/NixOS/nixpkgs/blob/c9b7cc79e6ce6b38debb28ab0d06051a4fb5b336/lib/trivial.nix#L106-L114) instead?

Comment (jtojnar) [reply pkgs/development/libraries/polkit/default.nix:36]: Thanks, was looking at the wrong library module so I did not find this.

File: pkgs/development/libraries/polkit/default.nix [hunk 2]
```diff
@@ -69,45 +140,39 @@ stdenv.mkDerivation rec {
...
+    rm --recursive "${DESTDIR}/etc"
+    rsync --archive "${DESTDIR}${system}"/* "$out"
+    rm --recursive "${DESTDIR}${system}"/*
+    rmdir --parents --ignore-fail-on-non-empty "${DESTDIR}${system}"
+    for o in $outputs; do
+        rsync --archive "${DESTDIR}/''${!o}" "$(dirname "''${!o}")"
+        rm --recursive "${DESTDIR}/''${!o}"
+    done
+    # Ensure the DESTDIR is removed.
+    rmdir "${DESTDIR}/nix/store" "${DESTDIR}/nix" "${DESTDIR}"
```
Comment (r-burns) [inline pkgs/development/libraries/polkit/default.nix:173]: Should probably use builtins.storeDir here

**Comment by jtojnar** [comment]
Looks like it is failing now for some reason. Weirdly, the error message points to a file that does not contain the include directive the message says.

```
[64/86] Compiling C object test/polkit/polkitidentitytest.p/polkitidentitytest.c.o-agent-1.so.0.0.0.symbols
FAILED: test/polkit/polkitidentitytest.p/polkitidentitytest.c.o 
gcc -Itest/polkit/polkitidentitytest.p -Itest/polkit -I../test/polkit -Isrc -I../src -Isrc/polkit -I/nix/store/150bwrv7p7qhb0vm0zwkw9z980gc58nq-glib-2.70.1-dev/include -I/nix/store/150bwrv7p7qhb0vm0zwkw9z980gc58nq-glib-2.70.1-dev/include/glib-2.0 -I/nix/store/5dckamm3ls6mgyp4p0nd8dvpv499m1a4-glib-2.70.1/lib/glib-2.0/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -DHAVE_CONFIG_H -D_POLKIT_COMPILATION -D_POLKIT_BACKEND_COMPILATION -MD -MQ test/polkit/polkitidentitytest.p/polkitidentitytest.c.o -MF test/polkit/polkitidentitytest.p/polkitidentitytest.c.o.d -o test/polkit/polkitidentitytest.p/polkitidentitytest.c.o -c ../test/polkit/polkitidentitytest.c
../test/polkit/polkitidentitytest.c:22:10: fatal error: config.h: No such file or directory
   22 | #include "config.h"
      |          ^~~~~~~~~~
compilation terminated.
[65/86] Compiling C object test/polkit/polkitunixnetgrouptest.p/polkitunixnetgrouptest.c.o
FAILED: test/polkit/polkitunixnetgrouptest.p/polkitunixnetgrouptest.c.o 
gcc -Itest/polkit/polkitunixnetgrouptest.p -Itest/polkit -I../test/polkit -Isrc -I../src -Isrc/polkit -I/nix/store/150bwrv7p7qhb0vm0zwkw9z980gc58nq-glib-2.70.1-dev/include -I/nix/store/150bwrv7p7qhb0vm0zwkw9z980gc58nq-glib-2.70.1-dev/include/glib-2.0 -I/nix/store/5dckamm3ls6mgyp4p0nd8dvpv499m1a4-glib-2.70.1/lib/glib-2.0/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -DHAVE_CONFIG_H -D_POLKIT_COMPILATION -D_POLKIT_BACKEND_COMPILATION -MD -MQ test/polkit/polkitunixnetgrouptest.p/polkitunixnetgrouptest.c.o -MF test/polkit/polkitunixnetgrouptest.p/polkitunixnetgrouptest.c.o.d -o test/polkit/polkitunixnetgrouptest.p/polkitunixnetgrouptest.c.o -c ../test/polkit/polkitunixnetgrouptest.c
../test/polkit/polkitunixnetgrouptest.c:22:10: fatal error: config.h: No such file or directory
   22 | #include "config.h"
      |          ^~~~~~~~~~
compilation terminated.
```

**Comment by jtojnar** [comment]
Turns out it was the musl patch breaking this. And while that can be trivially fixed using

```patch
--- a/test/polkit/meson.build
+++ b/test/polkit/meson.build
@@ -14,6 +14,7 @@ foreach test_unit: test_units
   exe = executable(
     test_unit,
     test_unit + '.c',
+    include_directories: top_inc,
     dependencies: libpolkit_gobject_dep,
     c_args: c_flags,
   )
```

It also breaks the tests so something is probably wrong with it cc @yu-re-ka

**Comment by yu-re-ka** [comment]
Just making it conditional again is fine with me. None of the musl distros have updated to 0.120 yet, and the patch hasn't been merged upstream.

**Comment by trofi** [comment]
`nixpkgs/staging` now fails to build at least:
- `accountsservice`
- `networkmanager`

Both fail in a similar way:

```
$ nix build -f. accountsservice networkmanager -L
...
accountsservice> install flags: install
accountsservice> [0/1] Installing files.
accountsservice> Installation failed due to insufficient permissions.
accountsservice> Attempting to use polkit to gain elevated privileges...
accountsservice> pkexec must be setuid root
accountsservice> FAILED: meson-install
...
networkmanager> Installation failed due to insufficient permissions.
networkmanager> Attempting to use polkit to gain elevated privileges...
networkmanager> pkexec must be setuid root
networkmanager> FAILED: meson-install
```

Bisect points at 527b98a125d381a1f03c2fa8393f80205c6dd052 `polkit: port to Meson`.

File: pkgs/development/libraries/polkit/default.nix [hunk 3]
```diff
@@ -12,22 +33,25 @@
...
+    repo = "polkit";
+    rev = version;
+    sha256 = "oEaRf1g13zKMD+cP1iwIA6jaCDwvNfGy2i8xY8vuVSo=";
   };
 
-  patches = [
+  patches = lib.optionals stdenv.hostPlatform.isMusl [
     # Make netgroup support optional (musl does not have it)
     # Upstream MR: https://gitlab.freedesktop.org/polkit/polkit/merge_requests/10
     # We use the version of the patch that Alpine uses successfully.
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/polkit/default.nix:57]: Is this causing any issues? If not I would like to keep it applied all the time that it does not bitrot.

Comment (jtojnar) [reply pkgs/development/libraries/polkit/default.nix:57]: Yes, it does, it breaks build and tests.

**Comment by jtojnar** [comment]
@trofi That is super weird, meson should only use polkit on permission error and then it would fail even previously because the polkit available in sandbox will not be setuid. Will investigate shortly or revert.

**Comment by jtojnar** [comment]
Should be fixed in 595c25f10535b4cde8b15109f8f36e387cce58c6 (and 3b916a6b78e41f488a02fd32c2f8b8816993f789, sorry, too tired).

**Comment by trofi** [comment]
Fixed all build failures for me. Thank you!

**Pull request merged**